src/Credentials.js

```diff
@@ -212,23 +212,18 @@ module.exports = class Credentials {
     return this._bearer_token_promise;
   }
 
-  async authorizationHeader(url) {
+  async authorizationHeader(url, method, body) {
     if (this.appAuth()) {
       await this.createBearerToken();
       return `Bearer ${this.bearer_token}`;
     }
-
+    let data = { url: url.toString(), method: method };
+    if (body != null) data.data = body;
     return this._oauth.toHeader(
-      this._oauth.authorize(
-        {
-          url: url.toString(),
-          method: 'get',
-        },
-        {
-          key: this.access_token_key,
-          secret: this.access_token_secret,
-        }
-      )
+      this._oauth.authorize(data, {
+        key: this.access_token_key,
+        secret: this.access_token_secret,
+      })
```

Comment:
NIT: for readability, I'd prefer to avoid using control flow here. This logic is largely declarative and can still be written inline cleanly.

```js
    return this._oauth.toHeader(
      this._oauth.authorize(
        {
          url: url.toString(),
          method,
          data: body || undefined,
        },
        {
          key: this.access_token_key,
          secret: this.access_token_secret,
        }
      )
    ).Authorization;
```